--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,6 +75,12 @@
     target_compile_options(_neon64 PRIVATE -march=armv8.2-a+dotprod+i8mm)
     list(APPEND ly_lib_parts $<TARGET_OBJECTS:_neon64>)
   endif()
+elseif(MSVC AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64")
+  list(APPEND ly_srcs 
+        \/compare_neon64.cc
+        \/rotate_neon64.cc
+        \/row_neon64.cc
+        \/scale_neon64.cc)
 endif()
 
 add_library( \ STATIC \ )
